compiler/src/dotty/tools/dotc/coverage/Location.scala

```diff
@@ -0,0 +1,43 @@
+package dotty.tools.dotc
+package coverage
+
+import ast.tpd._
+import dotty.tools.dotc.core.Contexts.Context
+import dotty.tools.dotc.core.Flags.*
+
+/** @param packageName
+  *   the name of the encosing package
+  * @param className
+  *   the name of the closes enclosing class
+  * @param fullClassName
+  *   the fully qualified name of the closest enclosing class
+  */
+final case class Location(
+    packageName: String,
+    className: String,
+    fullClassName: String,
+    classType: String,
+    method: String,
+    sourcePath: String
+)
+
+object Location:
+  def apply(tree: Tree)(using ctx: Context): Location =
+
+    val enclosingClass = ctx.owner.denot.enclosingClass
+    val packageName = ctx.owner.denot.enclosingPackageClass.name.toSimpleName.toString
+    val className = enclosingClass.name.toSimpleName.toString
+
+    val classType: String =
```

Comment:
How is this String used by scoverage?

Reply:
It's [serialized](https://github.com/lampepfl/dotty/blob/5cb5872ca4a5c83da5f5fc0a73b1d49b73d03751/compiler/src/dotty/tools/dotc/coverage/Serializer.scala#L65) in the scoverage report, for use by other tools.

Reply:
Right but I was wondering what these tools did with this information in practice, any idea?

Reply:
Honestly I'm not sure that it's actually used. I've just checked the HTML reports generated by sbt-coverage, and it doesn't display this information at all. For Sonarqube I'm not sure.